ttnn/cpp/ttnn/operations/creation.hpp

```diff
@@ -218,14 +218,26 @@ inline constexpr ZerosLike zeros_like{};
 inline constexpr OnesLike ones_like{};
 
 struct Empty {
-   static ttnn::Tensor invoke(
-    const ttnn::Shape& shape,
-    const DataType& dtype,
-    const Layout& layout,
-    Device* device,
-    const MemoryConfig& memory_config) {
+    static ttnn::Tensor invoke(
+        const ttnn::Shape& shape,
+        const DataType& dtype,
+        const Layout& layout,
+        Device* device,
+        const MemoryConfig& memory_config) {
         return create_device_tensor(shape, dtype, layout, device, memory_config);
     }
+
+    static ttnn::Tensor invoke(
+        const ttnn::Shape& shape,
+        const DataType& dtype,
+        const Layout& layout,
+        distributed::MeshDevice* device,
+        const MemoryConfig& memory_config) {
+        Tensor device_tensor = allocate_tensor_on_device(shape, dtype, layout, device, memory_config);
+        device_tensor.wait_for_tensor_metadata_populated();
+        device_tensor.wait_for_tensor_data_populated();
+        return device_tensor;
```

Comment:
@tt-asaigal what's the correct apis here to use. I don't think we want to call this in the worker thread?